pkg/daemon/update.go

```diff
@@ -381,8 +394,8 @@ func canonicalizeKernelType(kernelType string) string {
 
 // NewMachineConfigDiff compares two MachineConfig objects.
 func NewMachineConfigDiff(oldConfig, newConfig *mcfgv1.MachineConfig) *MachineConfigDiff {
-	oldIgn := oldConfig.Spec.Config
-	newIgn := newConfig.Spec.Config
+	oldIgn, _ := ctrlcommon.DecodeIgnitionConfigSpecV2(oldConfig.Spec.Config.Raw)
```

Comment:
is there a reason we aren't checking for the error here? (everywhere else there is a check when this func is called)

Reply:
I've changed up all functions that use this, including this one to also return an error now :)